build-tools/src/main/kotlin/org/jetbrains/kotlin/CompileCToBitcode.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin
+
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+import java.io.File
+import javax.inject.Inject
+
+open class CompileCToBitcode @Inject constructor(@InputDirectory val srcRoot: File,
```

Comment:
Are you sure we don't have this functionality already?

Reply:
There is similar for compiling C++. it's possible to combine both to one and make language as property. Should I combine?

Reply:
Yes